presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -679,9 +679,15 @@ protected RowExpression visitInPredicate(InPredicate node, Void context)
         {
             ImmutableList.Builder<RowExpression> arguments = ImmutableList.builder();
             arguments.add(process(node.getValue(), context));
-            InListExpression values = (InListExpression) node.getValueList();
-            for (Expression value : values.getValues()) {
-                arguments.add(process(value, context));
+            // InList of correlated sub-queries can be SymbolReference or other types
+            if (node.getValueList() instanceof InListExpression) {
+                InListExpression values = (InListExpression) node.getValueList();
+                for (Expression value : values.getValues()) {
+                    arguments.add(process(value, context));
+                }
+            }
+            else {
```

Comment:
as discussed offline, let's understand more regarding what `testDoubleNestedCorrelatedSubqueries` is trying to do

Reply:
I will let the test throw as it should for now. We need to implement correlated Projection to fully support this query, which might need more investigation.